visualization/app/codeCharta/core/scenario/scenario.service.ts

```diff
@@ -52,10 +49,23 @@ export class ScenarioService {
         }
     }
 
-    /**
-     * Returns an array of all scenarios.
-     * @returns {Scenario[]} all scenarios
-     */
+    private updateSettingsUsingScenario(settings, scenarioSettings): any {
```

Comment:
try using spread operator

Reply:
Works perfectly. It even works in nested levels 👍